src/charm.py

```diff
@@ -94,6 +99,15 @@ def _configure_pebble(self) -> None:
             self._container.add_layer(self._container_name, layer, combine=True)
             self._container.restart(self._service_name)
 
+    def _configure_upf_information(self):
+        """The `fiveg_n4` relation is not mandatory.
+
+        If it exists it must contain the UPF hostname and port.
+        """
+        if self.model.relations.get(FIVEG_N4_RELATION_NAME):
+            if not self._get_upf_hostname() or not self._get_upf_port():
```

Comment:
Why are we not looping over the relations here but doing it in the `_configure_gnb_information` method?